docs/no_peeking/mappings/biosample.py

```diff
@@ -0,0 +1,94 @@
+"""
+This module maps Kids First biospecimen to Phenopackets Biosample (derived from FHIR Specimen).
+Please visit https://aehrc.github.io/fhir-phenopackets-ig/StructureDefinition-Biosample.html
+    for the detailed structure definition.
+"""
+from kf_lib_data_ingest.common import constants
+from kf_lib_data_ingest.common.concept_schema import CONCEPT
+
+
+def biosample_status(x):
+    """
+    http://hl7.org/fhir/R4/valueset-specimen-status.html
+    """
+    if x == constants.COMMON.FALSE:
+        return 'unavailable'
+    elif x == constants.COMMON.TRUE:
+        return 'available'
+    else:
+        raise Exception('Unknown Biosample status')
+
+def biosample_type(x):
+    """
+    http://terminology.hl7.org/ValueSet/v2-0487
+    """
+    if x == constants.SPECIMEN.COMPOSITION.BLOOD:
+        return {
+            'system': 'http://terminology.hl7.org/CodeSystem/v2-0487',
+            'code': 'BLD',
+            'display': 'Whole blood'
+        }
+    elif x == constants.SPECIMEN.COMPOSITION.SALIVA:
+        return {
+            'system': 'http://terminology.hl7.org/CodeSystem/v2-0487',
+            'code': 'SAL',
+            'display': 'Saliva'
+        }
+    elif x == constants.SPECIMEN.COMPOSITION.TISSUE:
+        return {
+            'system': 'http://terminology.hl7.org/CodeSystem/v2-0487',
+            'code': 'TISS',
+            'display': 'Tissue'
+        }
+    else:
+        raise Exception('Unknown Biosample type')
+
+
+biosample = {
+    'resourceType': 'Specimen',
+    'id': CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID,
+    'meta': {
+        'profile': ['http://ga4gh.fhir.phenopackets/StructureDefinition/Biosample']
```

Comment:
We could have placeholder KF profiles based on the PK profiles to give us more flexibility down the road. We can also copy their "differential" element definitions to our KF profiles (so we have similar and interoperable constraints) and keep our profiles based on the base FHIR profiles.